plugins/woocommerce/src/Internal/DataStores/Orders/DataSynchronizer.php

```diff
@@ -74,6 +74,18 @@ function() {
 				$this->maybe_start_synchronizing_pending_orders();
 			}
 		);
+
+		// When posts is authoritative and sync is enabled, deleting a post also deletes COT data.
+		add_action(
+			'deleted_post',
+			function( $postid, $post ) {
+				if ( 'shop_order' === $post->post_type && ! $this->custom_orders_table_is_authoritative() && $this->data_sync_is_enabled() ) {
+					$this->data_store->delete_order_data_from_custom_order_tables( $postid );
```

Comment:
This `data_store` var is an instance of `OrdersTableDataStore` so it should have the method (unless I'm missing something). As whether to add it to the datastore itself, I have no preference, so I'd be happy to move it to the datastore if you think that's better.

My thinking was that it is sort of a "synchronization" task (propagating a delete from posts to COT tables) and that's why I chose `DataSynchronizer` for it to live as it could be possibly be part of the posts > COT migrator eventually.

Reply:
I see, makes sense this way too. I wanted to keep it in the data store since this is a direct query and ideally it would be good to have them all in the same place. Given that you have already moved it let's keep it as it is.

Reply:
Makes sense, let's keep it as it is.